docs/modules/ROOT/pages/api/upgrades.adoc

```diff
@@ -0,0 +1,86 @@
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+:class_hash: https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/class-hash/[class hash]
+
+= Upgrades
+
+Reference of interfaces and utilities related to upgradeability.
+
+== Core
+
+[.contract]
+[[IUpgradeable]]
+=== `++IUpgradeable++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/upgrades/interface.cairo#L3[{github-icon},role=heading-link]
+
+:Upgraded: xref:Upgradeable-Upgraded[Upgraded]
+
+```javascript
+use openzeppelin::upgrades::interface::IUpgradeable;
+```
+
+Interface of an upgradeable contract.
+
+[.contract-index]
+.Functions
+--
+* xref:#IUpgradeable-upgrade[`++upgrade(new_class_hash)++`]
+--
+
```

Comment:
This is definitely out of scope, but should we include the `Upgraded` event as part of the `IUpgradeable` interface? New SNIP, perhaps?

Reply:
I think I like the idea, but I would let it for another PR :)

Reply:
@andrew-fleming indeed the upgrade snip is due, and the event is already an integral part of the design!